can/exceptions.py

```diff
@@ -0,0 +1,61 @@
+"""
+There are several specific :class:`Exception` classes to allow user
+code to react to specific scenarios related to CAN busses::
+
+    Exception (Python standard library)
+     +-- ...
+     +-- CanError (python-can)
+         +-- CanBackendError
+         +-- CanInitializationError
+         +-- CanOperationError
+         +-- CanTimeoutError
+
+Keep in mind that some functions and methods may raise different exceptions.
+For example, validating typical arguments and parameters might result in a
+:class:`ValueError`. This should be documented for the function at hand.
+"""
+
+
+class CanError(Exception):
+    """Base class for all CAN related exceptions."""
+
+
+class CanBackEndError(CanError):
+    """Indicates an error related to the backend (e.g. driver/OS/library).
+
+    Example scenarios:
+      - The interface does not exist
+      - The interface is unsupported on the current platform
+      - The driver is not present or has the wrong version
+    """
+
+
+class CanInitializationError(CanError):
+    """Indicates an error the occurred while initializing a :class:`can.BusABC`.
+
+    Example scenarios:
+      - Try to open a non-existent device and/or channel
+      - Try to use an invalid setting, which is ok by value, but not ok for the interface
+      - The device or other resources are already used
+    """
+
+
+class CanOperationError(CanError):
+    """Indicates an error while in operation.
+
+    Example scenarios:
+      - A call to a library function results in an unexpected return value
+      - An invalid message was received
+      - Attempted to send an invalid message

```

Comment:
I assume that if python-can detects the error we raise a `ValueError`, but if the driver rejects it we would raise this?

Reply:
Yes. I guess it is sometimes hard to distinguish why sending a message caused an error, so I propose making this clearer in the docs.